File: 563. Binary Tree Tilt/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findTilt = function(root) {
  let tilt = 0;

  function visit(node) {
    if (!node) return 0;
    const left = visit(node.left);
    const right = visit(node.right);
    tilt += Math.abs(left - right);
    return left + right + node.val;
  }

  visit(root);
  return tilt;
};

File: 563. Binary Tree Tilt/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findTilt(root *TreeNode) int {
	totalTilt := 0

	var treeSum func(*TreeNode) int
	treeSum = func(n *TreeNode) int {
		if n == nil {
			return 0
		}

		l, r := treeSum(n.Right), treeSum(n.Left)
		tilt := l - r
		tilt += tilt >> 64 * 2 * tilt
		totalTilt += tilt
		return l + r + n.Val
	}

	treeSum(root)
	return totalTilt
}